File: software/examples/perl/example_threshold.pl
#!/usr/bin/perl

use Tinkerforge::IPConnection;
use Tinkerforge::BrickletLoadCell;

use constant HOST => 'localhost';
use constant PORT => 4223;
use constant UID => 'XYZ'; # Change XYZ to the UID of your Load Cell Bricklet

# Callback subroutine for weight reached callback
sub cb_weight_reached
{
    my ($weight) = @_;

    print "Weight: $weight g\n";
}

my $ipcon = Tinkerforge::IPConnection->new(); # Create IP connection
my $lc = Tinkerforge::BrickletLoadCell->new(&UID, $ipcon); # Create device object

$ipcon->connect(&HOST, &PORT); # Connect to brickd
# Don't use device before ipcon is connected

# Get threshold callbacks with a debounce time of 1 second (1000ms)
$lc->set_debounce_period(1000);

# Register weight reached callback to subroutine cb_weight_reached
$lc->register_callback($lc->CALLBACK_WEIGHT_REACHED, 'cb_weight_reached');

# Configure threshold for weight "greater than 200 g"
$lc->set_weight_callback_threshold('>', 200, 0);

print "Press key to exit\n";
<STDIN>;
$ipcon->disconnect();

File: software/examples/php/ExampleThreshold.php
<?php

require_once('Tinkerforge/IPConnection.php');
require_once('Tinkerforge/BrickletLoadCell.php');

use Tinkerforge\IPConnection;
use Tinkerforge\BrickletLoadCell;

const HOST = 'localhost';
const PORT = 4223;
const UID = 'XYZ'; // Change to your UID

// Callback function for weight reached callback (parameter has unit g)
function cb_weightReached($weight)
{
    echo "Weight: $weight g\n";
}

$ipcon = new IPConnection(); // Create IP connection
$lc = new BrickletLoadCell(UID, $ipcon); // Create device object

$ipcon->connect(HOST, PORT); // Connect to brickd
// Don't use device before ipcon is connected

// Get threshold callbacks with a debounce time of 1 second (1000ms)
$lc->setDebouncePeriod(1000);

// Register weight reached callback to function cb_weightReached
$lc->registerCallback(BrickletLoadCell::CALLBACK_WEIGHT_REACHED, 'cb_weightReached');

// Configure threshold for weight "greater than 200 g" (unit is g)
$lc->setWeightCallbackThreshold('>', 200, 0);

echo "Press ctrl+c to exit\n";
$ipcon->dispatchCallbacks(-1); // Dispatch callbacks forever

?>
